### PR TITLE
Handling embedded subProcess

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2735

### CODE REVIEW

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java [hunk 1]
```diff
@@ -630,10 +650,322 @@ public void testTwoSubProcesses() {
...
+        // Clean up
+        runtimeService.deleteProcessInstance(subProcessId1, "Clean up");
+        runtimeService.deleteProcessInstance(subProcessId2, "Clean up");
+
         runtimeService.deleteProcessInstance(processInstanceId, "Clean up");
 
     }
 
+    @Test
+    public void testEmbeddedSubProcesses() {
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java]: @CTI777, let's create another test suite to cover embedded subprocesses (`EmbeddedSubProcessAuditIT`). This class is becoming too big.

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java [hunk 2]
```diff
@@ -630,10 +650,322 @@ public void testTwoSubProcesses() {
...
+                                     tuple(TASK_UPDATED,processInstanceId, null, SIMPLE_EMBEDDED_SUB_PROCESS_WITH_CALLACTIVITY));
+        });
+
+        
+        //when
+        taskRestTemplate.complete(task);
+        
+        //Check we have two processes and one of them is callActivity process
+        // when
+        List<org.activiti.engine.runtime.ProcessInstance> processes = runtimeService.createProcessInstanceQuery()
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java]: Prefer to use the API than internal `RuntimeService`: you can use `ProcessInstanceRestTemplate.getSubprocesses(processInstanceId)` for this.

File: activiti-cloud-starter-runtime-bundle/src/test/resources/processes/activitySignalEventsInSubprocess.bpmn20.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    <userTask id="shipOrder" name="Wait" />
+
+    <sequenceFlow id="flow2" sourceRef="shipOrder" targetRef="theEnd" />
+
+    <endEvent id="theEnd" />
+
+ 	<sequenceFlow id="flow3" sourceRef="theStart" targetRef="eventSubProcess" />
+ 
+	<subProcess id="eventSubProcess" name="eventSubProcess" triggeredByEvent="true">
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/resources/processes/activitySignalEventsInSubprocess.bpmn20.xml]: You should remove `triggeredByEvent="true"`, as it's not triggered by an event anymore.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Some small concerns. See my comments bellow.

**Comment by CTI777** [comment]
@erdemedeiros @salaboy Another test added, subProcess with start message event
@erdemedeiros Refactoring made

**Review by erdemedeiros — APPROVED** [review]

**Review by salaboy — APPROVED** [review]

**Pull request merged**